flink/v1.19/flink/src/test/java/org/apache/iceberg/flink/source/TestIcebergSourceFailover.java

```diff
@@ -149,6 +163,15 @@ private void testBoundedIcebergSource(FailoverType failoverType) throws Exceptio
     JobClient jobClient = env.executeAsync("Bounded Iceberg Source Failover Test");
     JobID jobId = jobClient.getJobID();
 
+    Path jobCheckpointDir = checkpointDir.resolve(jobId.toString());
+    Awaitility.await("Wait for some checkpoints to complete")
```

Comment:
check for directory exist might be flaky. what if some subtask checkpointed its state and uploaded state file to the checkpoint dir but the whole checkpoint hasn't completed yet.

I can think of a couple potential alternatives

1. manually trigger checkpoint via `MiniCluster`

```
public CompletableFuture<String> triggerCheckpoint(JobID jobID) {
        return runDispatcherCommand(
                dispatcherGateway -> dispatcherGateway.triggerCheckpoint(jobID, rpcTimeout));
    }
```

2. check metrics like `TestIcebergSourceContinuous#assertThatIcebergSourceMetricExists()`

Reply:
Created a different test with creating a savepoint, and restarting from there